nvme.c

```diff
@@ -5380,12 +5388,12 @@ static int show_registers(int argc, char **argv, struct command *cmd, struct plu
 
 	_cleanup_nvme_dev_ struct nvme_dev *dev = NULL;
 	nvme_print_flags_t flags;
-	bool fabrics = false;
 	void *bar;
 	int err;
 
 	struct get_reg_config cfg = {
 		.human_readable	= false,
+		.fabrics = false,
```

Comment:
This change doesn't seem necessary. `nvme_get_properties` operates only on offsets. Anyway, the whole logic is already a bit complex so I'd rather avoid this type of additional info passed into the getters.

Reply:
Yes.
But in the nvme_get_single_property, If it is cfg.fabrics, check optional register.
Is there a better fix?

Thanks.

Reply:
I see, we already use it. This is so horrible code. I shouldn't have merged it.

Reply:
Okay. I understood it. 
As you said, it looks like refactoring is needed.